pkg/operator/controller/externaldns/deployment.go

```diff
@@ -265,8 +319,21 @@ func (r *reconciler) updateExternalDNSDeployment(ctx context.Context, current, d
 // Returns a boolean if an update is necessary, and the deployment resource to update to.
 func externalDNSDeploymentChanged(current, expected *appsv1.Deployment) (bool, *appsv1.Deployment) {
 	updated := current.DeepCopy()
+	return externalDNSSecretChanged(current, expected, updated) || externalDNSContainersChanged(current, expected, updated), updated
+}
 
-	return externalDNSContainersChanged(current, expected, updated), updated
+// externalDNSSecretChanged returns true if the current secret annotation differ from the expected
+func externalDNSSecretChanged(current, expected, updated *appsv1.Deployment) bool {
+	changed := false
+	for expected_key, expected_val := range expected.Annotations {
+		if current_val, ok := current.Annotations[expected_key]; ok {
+			if current_val != expected_val {
+				updated.Annotations[expected_key] = expected_val
+				return true
+			}
+		}
```

Comment:
Hi @DhritiShikhar 
If the annotation is not found, we should consider the secret as changed as well, right?

Reply:
Resolved here 578de6b1904c7a4cf790b72ef7f201ae6b04de65